spark_ec2.py

```diff
@@ -384,7 +387,7 @@ def validate_spark_hadoop_version(spark_version, hadoop_version):
         parts = spark_version.split(".")
         if parts[0].isdigit():
             spark_major_version = float(parts[0])
-            if spark_major_version > 1.0 and hadoop_version != "yarn":
+            if spark_major_version > 1.0 and "yarn" not in hadoop_version and "hadoop" not in hadoop_version:
```

Comment:
Can you add a regex check to make sure hadoop_version follows hadoop-x.x.x pattern?